### PR TITLE
New BindPlane OP Management console on GKE SecOps blueprint

### PR DESCRIPTION
**Pull request by simonebruzzechesse**
I applicable, I acknowledge that I have:
- [x] Read the [contributing guide](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md)
- [x] Ran `terraform fmt` on all modified files
- [x] Regenerated the relevant README.md files using [`tools/tfdoc.py`](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/CONTRIBUTING.md#fabric-tools)
- [x] Made sure all relevant tests pass

### CODE REVIEW

File: blueprints/secops/bindplane-gke/bindplane-deployment/versions_override.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+terraform {
+  required_providers {
+    restful = {
+      source = "magodo/restful"
```
Comment (juliocc) [inline blueprints/secops/bindplane-gke/bindplane-deployment/versions_override.tf:18]: hmmmm

I need to think about this

File: blueprints/secops/bindplane-gke/README.md [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+## Test
+
+```hcl
+module "bindplane-gke" {
+  source     = "./fabric/blueprints/secops/bindplane-gke"
+  project_id = "test"
+  bindplane_secrets = {
+    user            = "admin"
+    password        = "thisisnotasecret"
+    secret_key      = "xxxxxx-xxxxxxx-xxxxxx"
```
Comment (juliocc) [inline blueprints/secops/bindplane-gke/README.md:141]: This readme needs a warning stating that the state will contain these secrets.

File: blueprints/secops/bindplane-gke/README.md [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  source     = "./fabric/blueprints/secops/bindplane-gke"
+  project_id = "test"
+  project_create = {
+    billing_account_id = "12345-ABCDEF-12345"
+    parent             = "folders/2345678901"
+  }
+  bindplane_secrets = {
+    user            = "admin"
+    password        = "thisisnotasecret"
+    secret_key      = "xxxxxx-xxxxxxx-xxxxxx"
```
Comment (jsirianni) [inline blueprints/secops/bindplane-gke/README.md:141]: Secret key is a legacy option and no longer required. You can safely remove it.

On first login, the user's Organization and first Project will be created. The secret key for that project is generated during project creation and stored in the configured database (Postgres in the case of HA).

File: blueprints/secops/bindplane-gke/config/values.yaml.tpl [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+  # which is the bindplane namespace's bindplane service. This is suitable
+  # for connecting agents within the same cluster. We are using ingress
+  # so server_url needs to be updated to the ingress host.
+  # NOTE: server_url maps to bindplane's network.remoteURL option.
+  server_url: https://${hostname}
+
+
+# Enables mutli account, allowing you to create
+# multiple Tenants within the same BindPlane instance.
+multiAccount: true
```
Comment (jsirianni) [inline blueprints/secops/bindplane-gke/config/values.yaml.tpl:35]: This is a legacy option and can be safely removed. I understand it is in the upstream values and will make a ticket on our (observIQ) end to clean it up.

**Review by jsirianni — CHANGES_REQUESTED** [review]
I have not deployed this yet, but left a couple comments related to some legacy configuration options that no longer apply to BindPlane.

File: blueprints/secops/bindplane-gke/bindplane-deployment/versions_override.tf (hunk 1 above)
Comment (simonebruzzechesse) [reply blueprints/secops/bindplane-gke/bindplane-deployment/versions_override.tf]: Got rid of that, was not necessary

File: blueprints/secops/bindplane-gke/README.md (hunk 2 above)
Comment (simonebruzzechesse) [reply blueprints/secops/bindplane-gke/README.md]: Added warning section in the README file

File: blueprints/secops/bindplane-gke/README.md (hunk 3 above)
Comment (simonebruzzechesse) [reply blueprints/secops/bindplane-gke/README.md]: Ok got rid of that from the variables and chart, thanks for this input.

File: blueprints/secops/bindplane-gke/config/values.yaml.tpl (hunk 4 above)
Comment (simonebruzzechesse) [reply blueprints/secops/bindplane-gke/config/values.yaml.tpl]: Got rid of that, was just inherited from the official chart repository!

**Comment by simonebruzzechesse** [comment]
> I have not deployed this yet, but left a couple comments related to some legacy configuration options that no longer apply to BindPlane.

Please let me know if you are able to deploy it and any feedback on this setup, happy to support in case you have any issues with deployment! :)

File: blueprints/secops/bindplane-gke/versions.tf [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Fabric release: v36.0.1
+
+terraform {
```
[the remark this answers is not in the document]
Comment (simonebruzzechesse) [reply blueprints/secops/bindplane-gke/versions.tf]: Got rid of the versions.tf file!

File: blueprints/secops/bindplane-gke/bindplane-deployment/main.tf [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+  data = {
+    username        = var.bindplane_secrets.user
+    password        = local.bindplane_password
+    sessions_secret = var.bindplane_secrets.sessions_secret
+    license         = var.bindplane_secrets.license
+  }
+  type = "Opaque"
+}
+
+resource "kubernetes_secret" "tls" {
```
Comment (juliocc) [inline blueprints/secops/bindplane-gke/bindplane-deployment/main.tf:49]: I think we don't use this provider anyewhere else. If you have tests, you have to include at least one empty resource for `kubernetes`, `helm`, and `random` [here](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/tools/lockfile/main.tf)

**Review by juliocc — APPROVED** [review]

**Pull request merged**